packages/next/server/router.ts

```diff
@@ -399,8 +399,11 @@ export default class Router {
         }
 
         if (result.query) {
+          const remainingQuery = Object.entries(parsedUrlUpdated.query)
```

Comment:
This seems obscure, is there other way to fix it? Or maybe a comment would be nice. Also, we sure the internal parameters are all prefixed like this?

Reply:
I am not sure actually. Maybe someone from the Next team can let us know? I don't mind having a helper next to the actual type definition:

https://github.com/vercel/next.js/blob/8d1c58525c92308ccd10e830abca73c8f7e0c0d5/packages/next/server/request-meta.ts#L56-L64

Reply:
I like this idea of setting a helper that takes the same type. In fact, that code as it is right now would not consider `_nextDataReq`